File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    rank = {6: 1, 5: 2, 4: 3, 3: 4, 2: 5, 1: 6, 0: 6}
    win_count = 0
    zero_count = 0
    
    for i in lottos:
        if i in win_nums:
            win_count += 1
        elif i == 0:
            zero_count += 1
    
    return [rank[win_count + zero_count], rank[win_count]]

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    const rank = { 6: 1, 5: 2, 4: 3, 3: 4, 2: 5, 1: 6, 0: 6 };
    let win_count = 0;
    let zero_count = 0;
    
    for (let i of lottos) {
        if (win_nums.includes(i)) {
            win_count++;
        } else if (i === 0) {
            zero_count++;
        }
    }
    return [rank[win_count + zero_count], rank[win_count]];
}